src/sentry/middleware/integrations/parsers/base.py

```diff
@@ -212,8 +216,8 @@ def get_regions_from_organizations(
         """
         if not organizations:
             organizations = self.get_organizations_from_integration()
-        if not organizations:
-            logger.info("%s.no_organizations", self.provider, extra={"path": self.request.path})
-            return []
 
         return [get_region_for_organization(organization.slug) for organization in organizations]
+
+    def get_default_missing_integration_response(self) -> HttpResponse:
+        return HttpResponse(status=400)
```

Comment:
This should generally be "unreachable" for all integrations, and this should only run when an edge case was missed. We might want to add some logging here to surface this issue.

Reply:
I've been adding logging to each of the parsers before this is invoked, just because the behavior may differ depending on the integration itself. I'm starting to wonder if this is even the correct behavior, or if we should always just return a 202 if an org integration is missing, and returning a 500 + reporting the exception via our SDK if the parent integration is missing entirely since this can signify a misconfiguration on our end. What do you think?

Reply:
@GabeVillalobos I think we should still respond as a 400 status code, and then optionally log this as an exception.